### ISSUE TITLE
Flutter web rendering images with alpha channel with some artifacts

### ISSUE THREAD

**Issue by CustomSeif** [flutter/flutter#107801]
## Steps to Reproduce
**Option 1 **
You can run this code snippet using DartPad: [https://dartpad.dev/?id=a63c15e397943345b952d1c5513117c2](https://dartpad.dev/?id=a63c15e397943345b952d1c5513117c2)

**Option 2 **
Run the following code snippet with the device set to your web browser (I have tested both Chrome and Edge).

```
import "package:flutter/material.dart";

void main() {
    runApp(const MyApp());
}

class MyApp extends StatelessWidget {
    const MyApp({Key? key}) : super(key: key);

    @override
    Widget build(BuildContext context) {
        return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
                primarySwatch: Colors.blue,
            ),
            home: const Home(),
        );
    }
}

class Home extends StatelessWidget {
    const Home({ Key? key }) : super(key: key);

    final String imageURL = "https://i.postimg.cc/8PjMYGfZ/products-asics-1201a147-002-3e277fa0-a543-47e5-bd72-17403e91675d-700.png";

    @override
    Widget build(BuildContext context) {
        return Scaffold(
            body: Container(
                height: double.infinity,
                width: double.infinity,
                color: Colors.pink,
                child: Center(
                    child: Image.network(imageURL)
                )
            )
        );
    }
}
```

The versions in which I tested this issue are: `3.0.4`  `3.0.5`  `3.1.0-0.0.pre.941`

**Expected results:**

This is a screenshot of the image being rendered in a Windows desktop application. The image is being rendered correctly, which it also does do on android.

![](https://i.postimg.cc/t4rdTzn4/Screenshot-2022-07-17-135834.png)

**Actual results:**

When running the application on the Web (which I tested in both Chrome and Edge) the image has some white artifacts surrounding it.

![](https://i.postimg.cc/LsntZZjB/Screenshot-2022-07-17-135720.png)

**Comment by danagbemava-nc** [comment]
Hi @CustomSeif, it would appear that this issue has been fixed on the latest `master` channel. See the screenshots below

Can you upgrade to the latest version of master to see if you're still able to reproduce this issue?

<details>
<summary>screenshots</summary>


| stable | master | 
| -- | -- |
| <img width="883" alt="Screenshot 2022-07-18 at 10 36 20" src="https://user-images.githubusercontent.com/88313112/179497186-6414a23d-d815-488c-a321-e210c6b7a383.png"> | <img width="883" alt="Screenshot 2022-07-18 at 10 40 38" src="https://user-images.githubusercontent.com/88313112/179497199-2de6c5ff-51a8-493f-b032-dbfa237ba12d.png"> |

</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-GB)
    • Flutter version 3.0.5 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.69.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1700, on macOS 12.4 21F79 darwin-arm, locale en-GB)
    • Flutter version 3.1.0-0.0.pre.1700 on channel master at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2f0a25257e (2 hours ago), 2022-07-18 02:34:06 -0400
    • Engine revision 7186455dda
    • Dart version 2.19.0 (build 2.19.0-5.0.dev)
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F100
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.69.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by CustomSeif** [comment]
Hello @danagbemava-nc, I have switched over to the `master` branch and ran `flutter upgrade`. I can confirm it has been fixed, the version I ran this on is `3.1.0-0.0.pre.1704`
Thanks.
